sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -678,6 +678,14 @@ public abstract partial class TokenCredential
         public string[] Scopes { get { throw null; } }
         public string? TenantId { get { throw null; } }
     }
+    public partial class UserAgentValue
```

Comment:
There were comments in the APIView on the name of the type ... have you considered change it?

Reply:
I've considered it, but I haven't heard an alternative that everyone likes better.

Reply:
I liked the suggestion to make it more general than just user agent:
![image](https://user-images.githubusercontent.com/21269347/156848538-539f7c99-5e53-4d5c-a7a1-e9d99736a40f.png)

Reply:
It's not more general than that though. It is explicitly intended to encapsulate a properly formatted UserAgent string.

Reply:
You are creating an [attractive nuisance](https://en.wikipedia.org/wiki/Attractive_nuisance_doctrine) with a name like `UserAgentValue`.  The only people who should be using this type are library authors who want to customize their telemetry according to the Guidelines, not customers who want to change their `User-Agent` header to something random.

Reply:
Ok, I went with `TelemetryDetails`